### ISSUE TITLE
Nested populate array and spaces in path don't work together

### ISSUE THREAD

**Issue by AbdelrahmanHafez** [Automattic/mongoose#6284]
**Do you want to request a *feature* or report a *bug*?** Report a bug.
The following code would populate the cityId & districtId fields succesfully, _with only using the space between the two fields_.

```
Post.find({})
  .populate({
    path: 'userId',
    populate: {
      path: 'cityId districtId',
      select: 'name'
    }
  });
```

The following would populate the `houseId` succesfully, and would *only* populate the `cityId` field, would not populate both `cityId` & `districtId` as expected.

```
Post.find({})
  .populate({
    path: 'userId',
    populate: [{
      path: 'houseId',
      select: 'location'
    }, {
      path: 'cityId districtId',
      select: 'name'
    }]
  });
```

Node v8.9.3
Mongoose v5.0.1
MongoDB v3.2.7

**Comment by lineus** [comment]
confirmed with the following repro script:

### 6284.js
```
#!/usr/bin/env node
'use strict'

const assert = require('assert')
const mongoose = require('mongoose')
mongoose.connect('mongodb://localhost/test')
const Schema = mongoose.Schema

const houseSchema = new Schema({
  location: String
})

const userSchema = new Schema({
  name: String,
  houseId: {
    type: Schema.Types.ObjectId,
    ref: 'house'
  },
  cityId: {
    type: Schema.Types.ObjectId,
    ref: 'city'
  },
  districtId: {
    type: Schema.Types.ObjectId,
    ref: 'district'
  }
})

const citySchema = new Schema({
  name: String
})

const districtSchema = new Schema({
  name: String
})

const postSchema = new Schema({
  content: String,
  userId: {
    type: Schema.Types.ObjectId,
    ref: 'user'
  }
})

const House = mongoose.model('house', houseSchema)
const User = mongoose.model('user', userSchema)
const City = mongoose.model('city', citySchema)
const District = mongoose.model('district', districtSchema)
const Post = mongoose.model('post', postSchema)

const house = new House({ location: '123 abc st.' })
const city = new City({ name: 'Somecity' })
const district = new District({ name: 'That District' })

const user = new User({
  name: 'Billy',
  houseId: house._id,
  districtId: district._id,
  cityId: city._id
})

const post = new Post({
  content: 'Some meaningful insight.',
  userId: user._id
})

function cleanUp () {
  let colls = [ City, District, Post, User, House ]
  colls.forEach(async (Model) => {
    await Model.remove({})
  })
}

async function run () {
  await cleanUp()
  await House.create(house)
  await City.create(city)
  await District.create(district)
  await User.create(user)
  await Post.create(post)

  let firstPop = await Post.findOne().populate({
    path: 'userId',
    populate: {
      path: 'cityId districtId',
      select: 'name'
    }
  })

  assert.ok(firstPop.userId.cityId.name, '1) City Name Missing')
  assert.ok(firstPop.userId.districtId.name, '1) District Name Missing')

  let secondPop = await Post.findOne().populate({
    path: 'userId',
    populate: [{
      path: 'houseId',
      select: 'location'
    }, {
      path: 'cityId districtId',
      select: 'name'
    }]
  })

  assert.ok(secondPop.userId.houseId.location, '2) House Location Missing')
  assert.ok(secondPop.userId.cityId.name, '2) City Name Missing')
  assert.ok(secondPop.userId.districtId.name, '2) District Name Missing')
}

run().catch(console.error)
```
### Output:
```
issues: ./6284.js 
{ AssertionError [ERR_ASSERTION]: 2) District Name Missing
    at run (/Users/lineus/dev/Help/mongoose5/issues/6284.js:106:10)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: undefined,
  expected: true,
  operator: '==' }
^C
issues: 
```

**Comment by lineus** [comment]
@AbdelrahmanHafez I wasn't able to find any documentation that sets the expectation that the options object passed into .populate() should accept an array as the value of the populate parameter. `.populate({ path: 'somepath',  populate: [{..}] })`

I think this is a **very reasonable** feature request as opposed to a bug.

In the meantime ( while this issue is being addressed ), you can chain two populate calls together. Here is a working version of the second populate call from my example above, using chained populate calls:
```
let secondPop = await Post.findOne()
    .populate({
      path: 'userId',
      populate: {
        path: 'houseId',
        select: 'location -_id'
      }
    }).populate({
      path: 'userId',
      populate: {
        path: 'cityId districtId',
        select: 'name -_id'
      }
    })
```

**Comment by AbdelrahmanHafez** [comment]
I haven't looked for it in the documentation, but it sure works (except for the spaces case).
I would think that the double `.populate({ path: 'userId' })` Would look for the userId twice, to do the population, as opposed to the array, where it only finds it once.

**Comment by lineus** [comment]
yeah, it's the array **and** the space separated value of path that cause it to not work as you expect. The fact that it works with an array at all appears to me to be a side effect of the implementation of the internal utils.populate method, as opposed to a documented feature it provides.

If I can sort out a reasonable fix for it I'll put in a PR to add the feature.

**Comment by vkarpov15** [comment]
Yeah this is definitely a feature to add. As a workaround, do the below.

```javascript
Post.find({})
  .populate({
    path: 'userId',
    populate: [{
      path: 'houseId',
      select: 'location'
    }, {
      path: 'cityId',
      select: 'name'
    }, {
      path: 'districtId',
      select: 'name'
    }]
  });
```

**Comment by vkarpov15** [comment]
Fixed by #6296
